src/vcpkg/base/downloads.cpp

```diff
@@ -936,6 +936,10 @@ namespace vcpkg
                                  msg::url = replace_secrets(read_url, m_config.m_secrets));
                     return read_url;
                 }
+                else
+                {
+                    msg::println(msgAssetCacheMissBlockOrigin, msg::path = download_path.filename());
```

Comment:
This does not seem correct to me, can you show how we know x-block-origin is set here?

Reply:
OK, now it attempts the asset download, if it fails, we check if x-block-origin is enabled and print the relevant message.